### PR TITLE
BACKLOG-35892 Move Oracle bulk loader from core to plugin

### PR DESCRIPTION
**Pull request by GopiKrishnaAngadi**

### CODE REVIEW

File: plugins/oracle-bulk-loader/impl/src/main/java/org/pentaho/di/trans/OraBulkDataOutput.java [hunk 1]
```diff
@@ -20,7 +20,7 @@
  *
  ******************************************************************************/
 
-package org.pentaho.di.trans.steps.orabulkloader;
+package org.pentaho.di.trans;
```
Comment (peterrinehart) [inline plugins/oracle-bulk-loader/impl/src/main/java/org/pentaho/di/trans/OraBulkDataOutput.java:23]: Why did we need to move this into a different package?

Comment (GopiKrishnaAngadi) [reply plugins/oracle-bulk-loader/impl/src/main/java/org/pentaho/di/trans/OraBulkDataOutput.java]: changed the packaged and recommited

File: plugins/oracle-bulk-loader/impl/src/test/java/org/pentaho/di/trans/steps/orabulkloader/OraBulkDataOutputTest.java [hunk 2]
```diff
@@ -35,14 +35,9 @@
...
 import java.io.IOException;
-
 import static org.junit.Assert.assertTrue;
 import static org.junit.Assert.fail;
-import static org.mockito.Mockito.doReturn;
-import static org.mockito.Mockito.doThrow;
-import static org.mockito.Mockito.mock;
-import static org.mockito.Mockito.spy;
-import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.*;
```
Comment (peterrinehart) [inline plugins/oracle-bulk-loader/impl/src/test/java/org/pentaho/di/trans/steps/orabulkloader/OraBulkDataOutputTest.java:40]: Don't collapse imports with .*
We should explicitly import every class being used.

File: plugins/oracle-bulk-loader/impl/src/test/java/org/pentaho/di/trans/steps/orabulkloader/OraBulkLoaderMetaTest.java [hunk 3]
```diff
@@ -1,175 +1,171 @@
...
+import org.pentaho.di.core.exception.KettleException;
+import org.pentaho.di.core.plugins.PluginRegistry;
+import org.pentaho.di.junit.rules.RestorePDIEngineEnvironment;
+import org.pentaho.di.trans.steps.loadsave.LoadSaveTester;
+import org.pentaho.di.trans.steps.loadsave.validator.ArrayLoadSaveValidator;
+import org.pentaho.di.trans.steps.loadsave.validator.DatabaseMetaLoadSaveValidator;
+import org.pentaho.di.trans.steps.loadsave.validator.FieldLoadSaveValidator;
+import org.pentaho.di.trans.steps.loadsave.validator.StringLoadSaveValidator;
+
+import java.util.*;
```
Comment (peterrinehart) [inline plugins/oracle-bulk-loader/impl/src/test/java/org/pentaho/di/trans/steps/orabulkloader/OraBulkLoaderMetaTest.java:39]: Don't collapse imports with .*

File: plugins/oracle-bulk-loader/impl/src/test/java/org/pentaho/di/trans/steps/orabulkloader/OraBulkLoaderTest.java [hunk 4]
```diff
@@ -22,11 +22,7 @@
 package org.pentaho.di.trans.steps.orabulkloader;
 
 import org.apache.poi.util.TempFile;
-import org.junit.After;
-import org.junit.Before;
-import org.junit.BeforeClass;
-import org.junit.ClassRule;
-import org.junit.Test;
+import org.junit.*;
```
Comment (peterrinehart) [inline plugins/oracle-bulk-loader/impl/src/test/java/org/pentaho/di/trans/steps/orabulkloader/OraBulkLoaderTest.java:25]: Don't collapse imports with .*

**Review by peterrinehart — CHANGES_REQUESTED** [review]

File: plugins/oracle-bulk-loader/impl/src/test/java/org/pentaho/di/trans/steps/orabulkloader/OraBulkLoaderMetaTest.java (hunk 3 above)
Comment (GopiKrishnaAngadi) [reply plugins/oracle-bulk-loader/impl/src/test/java/org/pentaho/di/trans/steps/orabulkloader/OraBulkLoaderMetaTest.java]: done

File: plugins/oracle-bulk-loader/impl/src/test/java/org/pentaho/di/trans/steps/orabulkloader/OraBulkLoaderTest.java (hunk 4 above)
Comment (GopiKrishnaAngadi) [reply plugins/oracle-bulk-loader/impl/src/test/java/org/pentaho/di/trans/steps/orabulkloader/OraBulkLoaderTest.java:25]: done

File: plugins/pom.xml [hunk 5]
```diff
@@ -75,10 +75,9 @@
         <module>pur</module>
         <module>ms-access</module>
         <module>postgresql-db-bulk-loader</module>
+        <module>oracle-bulk-loader</module>
         <module>xml</module>
         <module>streaming</module>
-        <module>metastore-locator</module>
```
Comment (peterrinehart) [inline plugins/pom.xml:80]: I think you need to rebase one last time; we merged in more changes and these two modules should not get removed from this assembly here.

Comment (GopiKrishnaAngadi) [reply plugins/pom.xml:80]: done

**Review by peterrinehart — APPROVED** [review]

**Pull request merged**